### PR TITLE
added warning for reserved pin ids

### PR DESCRIPTION
**Pull request by fisher-alice**
## Summary

- Added a warning message in Debug Console and gutter warning when a reserved data pin is used as a parameter for Circuit Playground Maker commands. 

- Although there are 8 available pins, only 5 of them are actually available for use. 
Allowed data pins: A0, A1, A4, A5, A6 (Express) and 0, 2, 3, 6, 12 (Classic)
Reserved data pins: A2, A3, A7 (Express) and 1, 9, 10 (Classic)

- The warning message appears only if user uses any of the reserved data pins. 

Before updates:

- Circuit Playground Classic: pins 0, 11 are usable pin ids and 1, 9, 10 are reserved pin ids but no warning for reserved pins

Screen Shot 2022-06-30 at 7 25 21 PM

- Circuit Playground Express: pins A0, A6 are usable pin Ids and pins A2, A7 are reserved ids but no warning for reserved pins
Screen Shot 2022-06-30 at 7 00 20 PM

After updates:

- Circuit Playground Classic: pins 0, 11 are usable pin ids and 1, 9, 10 are reserved pin ids
Screen Shot 2022-06-30 at 6 47 59 PM

- Circuit Playground Express: pins A0, A6 are usable pin Ids and pins A2, A7 are reserved ids 
Screen Shot 2022-06-30 at 6 48 49 PM

## Links

[jira ticket](https://codedotorg.atlassian.net/browse/STAR-1706)

## Follow up work
I observed that there is an error message thrown when invalid parameters are used such as "A8" or 30.

image

image

In other words, any string other than "A0" through "A7" and any number not within range of 0 and 30 will result in an error displayed. However, there is no error thrown when an invalid number parameter is used within range of 0 and 30 such as 25 which is not a valid pin id. 

image

Follow up task could be to make sure an error is also thrown for data pin numbers other than the valid data pins (0, 1, 2, 3, 6, 9, 10, 12).

### CODE REVIEW

File: apps/src/lib/util/javascriptMode.js [hunk 1]
```diff
@@ -47,6 +47,8 @@ export function apiValidateType(
   const validatedTypeKey = 'validated_type_' + varName;
   if (typeof opts[validatedTypeKey] === 'undefined') {
     var properType;
+    var customWarning;
+    var warningMessage;
```
Comment (sanchitmalhotra126) [inline apps/src/lib/util/javascriptMode.js]: nit: not sure if we need to declare `warningMessage` up here if it's only used when we actually output the warning (unless we're using it somewhere else that I'm not seeing?)

**Review by sanchitmalhotra126 — COMMENTED** [review]
Code looks good to me, nice work! Looks like you may have to fix a unit test though - the Drone build has a failing unit test that has something to do with `pinmode` (see line 8596 in the Drone logs). I think this referencing a test in `kits/maker/commandsTest.js` so you may just want to run those locally to see what's going on.

File: apps/src/lib/util/javascriptMode.js (hunk 1 above)
Comment (fisher-alice) [reply apps/src/lib/util/javascriptMode.js]: Thanks! Updated scope of variable. Ran failed unit test locally and found the issue - the test used a reserved pin id (1) which should now fail since no longer can be used - I changed test to use valid pin id (0). I will now work on adding a unit test to test a Maker command with a reserved pin id.

**Review by epeach — APPROVED** [review]
LGTM!
